contracts/utils/PendingOwnableUpgradeable.sol

```diff
@@ -0,0 +1,166 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/utils/introspection/IERC165.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+
+import "../interfaces/IPendingOwnable.sol";
+import "./PendingOwnableErrors.sol";
+
+/**
+ * @title Pending Ownable
+ * @author Trader Joe
+ * @notice Contract module which provides a basic access control mechanism, where
+ * there is an account (an owner) that can be granted exclusive access to
+ * specific functions. The ownership of this contract is transferred using the
+ * push and pull pattern, the current owner set a `pendingOwner` using
+ * {setPendingOwner} and that address can then call {becomeOwner} to become the
+ * owner of that contract. The main logic and comments comes from OpenZeppelin's
+ * Ownable contract.
+ *
+ * By default, the owner account will be the one that deploys the contract. This
+ * can later be changed with {setPendingOwner} and {becomeOwner}.
+ *
+ * This module is used through inheritance. It will make available the modifier
+ * `onlyOwner`, which can be applied to your functions to restrict their use to
+ * the owner
+ */
+abstract contract PendingOwnableUpgradeable is
+    Initializable,
+    IERC165,
+    IPendingOwnable
```

Comment:
Same, use Upgradeable dependencies,
Btw it should use this https://github.com/OpenZeppelin/openzeppelin-contracts-upgradeable/blob/master/contracts/utils/introspection/ERC165Upgradeable.sol instead of the interface.